pages/admin/settings.tsx

```diff
@@ -0,0 +1,230 @@
+                        {headerGroup.headers.map((column) => (
+                            <th
+                                {...column.getHeaderProps()}
+                                style={{
+                                    borderBottom: "solid 3px gainsboro",
+                                    textAlign: "left",
+                                    paddingBottom: "10px",
+                                    fontFamily: "Geogrotesque Bold",
+                                }}
+                            >
+                                {column.render("Header")}
+                            </th>
+                        ))}
+                    </tr>
+                ))}
+            </thead>
+            <tbody
+                {...getTableBodyProps()}
+                style={{
+                    borderBottom: "solid 2px gainsboro",
+                }}
+            >
+                {rows.map((row) => {
+                    prepareRow(row);
+                    return (
+                        <tr {...row.getRowProps()}>
+                            {row.cells.map((cell) => {
+                                return (
+                                    <td
+                                        {...cell.getCellProps()}
+                                        style={{
+                                            padding: "10px 0px",
+                                        }}
+                                    >
+                                        {cell.render("Cell")}
+                                    </td>
+                                );
+                            })}
+                        </tr>
+                    );
+                })}
+            </tbody>
+        </table>
+    );
+
+    return (
+        <Flex direction="column" padding="16">
+            {modalType == ModalType.ADD && (
+                <AddModal
+                    isOpen={isOpen}
+                    onConfirm={() => {
+                        if (!email) {
+                            setEmailError("Please enter an email.");
+                            return;
+                        }
+                        console.log(
+                            `add admin with email: ${email}, role: ${role}`,
+                        );
+                        // TODO make request to back end
+                        onClose();
+                    }}
+                    onCancel={onClose}
+                    setEmail={setEmail}
+                    setRole={setRole}
+                    emailError={emailError}
+                />
+            )}
+            {modalType == ModalType.EDIT && (
+                <EditModal
+                    isOpen={isOpen}
+                    onConfirm={() => {
+                        if (!email) {
+                            setEmailError("Please enter an email.");
+                            return;
+                        }
+                        console.log(
```

Comment:
Is the idea to call the `GET` endpoint again after the add action? Might as well just update the data locally

Reply:
discussed: will either return the updated user list from POST/DELETE requests OR have the front end call a GET request after every POST/DELETE